### PR TITLE
[android] - fix black screen

### PR DESCRIPTION
**Pull request by tobrun**
This fixes the black screen issue we have seen on multiple devices. It reimplements a fix we used to have for these devices but was removed during the lift for rendering on main thread. 

👀  @tmpsantos 

cc @bleege @zugaldia @cammace @danswick

### CODE REVIEW

**Comment by tobrun** [comment]
@tmpsantos  indicated we could remove this line:

```
        oldDisplay = eglGetCurrentDisplay();
```

Since it will be filled up in activate();

**Comment by tmpsantos** [comment]
> @tmpsantos indicated we could remove the following lines:
> 
>         oldDisplay = eglGetCurrentDisplay();
>         oldReadSurface = eglGetCurrentSurface(EGL_READ);
>         oldDrawSurface = eglGetCurrentSurface(EGL_DRAW);
>         oldContext = eglGetCurrentContext();
> Since it will be filled up in activate();
> 
> update: I tried removing those but this results in a black screen again

It is the other way around. You keep the onde on the `createSurface()` and remove on `activate()`. :smile:

**Comment by tobrun** [comment]
> It is the other way around. You keep the onde on the createSurface() and remove on activate(). :smile:

:face_palm:

**Comment by tobrun** [comment]
@tmpsantos 
Tested it out and it works 🙇

File: platform/android/src/native_map_view.cpp [hunk 1]
```diff
@@ -127,11 +127,6 @@ std::array<uint16_t, 2> NativeMapView::getFramebufferSize() const {
 void NativeMapView::activate() {
     mbgl::Log::Debug(mbgl::Event::Android, "NativeMapView::activate");
 
-    oldDisplay = eglGetCurrentDisplay();
```
Comment (jfirebaugh) [inline platform/android/src/native_map_view.cpp]: This doesn't look correct to me -- we want `deactivate` to restore the display and surfaces that were active when `activate` was called. We shouldn't assume that's the same display and surfaces that were active when `createSurface` was called.

I think rather what we should do is call `activate` / `deactivate` _inside_ `createSurface`, around the portion of the code where we want our context active.

**Comment by tobrun** [comment]
@jfirebaugh I implemented the fix from https://github.com/mapbox/mapbox-gl-native/pull/4892#discussion_r61494850 in 13d1149:

This results in:
  - resolving black screen on Kazam Trooper
  - not introducing regressions on Nexus 6P or Nexus 5X (x86)

Could you validate if this was the solution you had in mind? 

cc @tmpsantos

**Comment by jfirebaugh** [comment]
Perfect. 🚢

**Comment by tmpsantos** [comment]
:shipit:

**Comment by tobrun** [comment]
Co-workers approved, CI approved, going to rebase and merge 🚀

**Comment by bleege** [comment]
\o/  Great job @tobrun!  🚀

**Pull request merged**